Source/WTF/wtf/cocoa/NSURLExtras.mm

```diff
@@ -136,53 +136,22 @@ static String decodePercentEscapes(const String& string)
     return result ? result.bridgingAutorelease() : URL;
 }
 
-NSURL *URLWithData(NSData *data, NSURL *baseURL)
-{
-    if (!data)
-        return nil;
-    
-    size_t length = [data length];
-    if (length > 0) {
-        // Work around <rdar://4470771>: CFURLCreateAbsoluteURLWithBytes(.., TRUE) doesn't remove non-path components.
-        baseURL = URLByTruncatingOneCharacterBeforeComponent(baseURL, kCFURLComponentResourceSpecifier);
-
-        const UInt8 *bytes = static_cast<const UInt8*>([data bytes]);
-
-        // CFURLCreateAbsoluteURLWithBytes would complain to console if we passed a path to it.
-        if (bytes[0] == '/' && !baseURL)
-            return nil;
-
-        // NOTE: We use UTF-8 here since this encoding is used when computing strings when returning URL components
-        // (e.g calls to NSURL -path). However, this function is not tolerant of illegal UTF-8 sequences, which
-        // could either be a malformed string or bytes in a different encoding, like shift-jis, so we fall back
-        // onto using ISO Latin 1 in those cases.
-        auto result = adoptCF(CFURLCreateAbsoluteURLWithBytes(nullptr, bytes, length, kCFStringEncodingUTF8, (__bridge CFURLRef)baseURL, YES));
-        if (!result)
-            result = adoptCF(CFURLCreateAbsoluteURLWithBytes(nullptr, bytes, length, kCFStringEncodingISOLatin1, (__bridge CFURLRef)baseURL, YES));
-        return result.bridgingAutorelease();
-    }
-    return [NSURL URLWithString:@""];
-}
-
 static NSData *dataWithUserTypedString(NSString *string)
 {
     NSData *userTypedData = [string dataUsingEncoding:NSUTF8StringEncoding];
     ASSERT(userTypedData);
     
     const UInt8* inBytes = static_cast<const UInt8 *>([userTypedData bytes]);
-    int inLength = [userTypedData length];
-    if (!inLength)
-        return nil;
+    NSUInteger inLength = [userTypedData length];
 
-    CheckedInt32 mallocLength = inLength;
-    mallocLength *= 3; // large enough to %-escape every character
-    if (mallocLength.hasOverflowed())
+    // large enough to %-escape every character
+    if (!inLength || inLength > (INT32_MAX / 3))
```

Comment:
Why get rid of the CheckedArithmetic logic? Seems like you've reimplemented it by hand.